src/newComponents/viewer/LeadPreview/Preview/index.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import {
+    _cs,
+    isValidUrl,
+} from '@togglecorp/fujs';
+import {
+    ImagePreview,
+} from '@the-deep/deep-ui';
+
+import { isHttps } from '#utils/safeCommon';
+
+import {
+    MimeTypes,
+    isImageMimeType,
+    isDocMimeType,
+    isHTMLMimeType,
+    createUrlForGoogleViewer,
+} from './mimeTypes';
+
+import styles from './styles.scss';
+
+interface Props {
+    className?: string;
+    url: string;
+    mimeType: MimeTypes;
+    canShowIframe: boolean;
+}
+
+function Preview(props: Props) {
+    const {
+        className,
+        url,
+        mimeType,
+        canShowIframe,
+    } = props;
+
+    const isRunningInHTTP = window.location.protocol === 'http:';
+    const isCurrentUrlHTTPs = isHttps(url);
+
+    // NOTE: We are checking if current URL is https or the system is running in
+    // http mode because http url shouldn't be embedded under http env
+    if (isImageMimeType(mimeType) && (isCurrentUrlHTTPs || isRunningInHTTP)) {
```

Comment:
We need to make this logic clearer and more correct.

Reply:
Lets discuss this over a call